### PR TITLE
Adds configuration for motion and tap detection features, and some refactoring

### PR DESCRIPTION
**Pull request by kyp44**
Hi, thanks for writing this driver, it's saved me a lot of time on a project I'm currently working on!

I needed to add some methods to configure the motion detection and tap detection features, to configure the interrupt pins, and map these features to the interrupts. I've tested all this and have them working well in my project. I also thought the crate could use some refactoring to make things a bit more efficient when configuring features, and to make the driver a little more robust in terms of tracking its state and not allowing bad things to happen. Please refer to the commit log and documentation for more details.

Also, I went ahead and added at least some documentation for every item and method, noting that bit mask items can evidently not be documented due to the macro expansion.

Let me know if you take issue with any of the design decisions I made, or have any ways to improve them.

I'm also interested in completing the implemented features down the road once I get my project to a certain point (the features I've already implemented are all I need for this project).

Thanks for considering this request and don't hesitate to provide any feedback on the changes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 [dependencies]
-embedded-hal = "^0.2"
-num_enum = { version = "^0.5", default-features = false }
-accelerometer = { version = "^0.12", optional = true }
-bitmask-enum = { version = "^1.1" }
+accelerometer = {version = "^0.12", optional = true}
+bitmask-enum = {version = "^1.1"}
+derive-new = "0.5"
```
Comment (pyaillet) [inline Cargo.toml]: Is this a strong requirement on your side?

I am not a fan of introducing new dependencies if it can be avoided.

**Review by pyaillet — COMMENTED** [review]
Hi!

Thank you very much for your interest and your contribution :)

File: Cargo.toml (hunk 1 above)
Comment (kyp44) [reply Cargo.toml]: It looks like the only actual new dependency I added is `derive-new`, noting that I also updated the version of `embedded-hal`. The new dependency is just basically a macro that allows automatically deriving a constructor method (i.e. a `new` function). This is definitely not a strong requirement at all and I have no problem with removing it in the interest of minimizing dependencies. It's only used in one place, on the `Config` struct, and can be easily replaced with a trivial manual `new` constructor. This struct also has all public fields, so maybe doesn't really even need a constructor probably.

Comment (pyaillet) [reply Cargo.toml]: Yes, my concern was only about `derive-new`.
I agree that for the `Config` struct we could get rid of the constructor and rely on the public fields.

Comment (kyp44) [reply Cargo.toml]: Added a new commit that removes the `derive-new` dependency.

**Comment by pyaillet** [comment]
Thank you!

**Pull request merged**